.circleci/config.yml

```
@@ -582,6 +587,71 @@ jobs:
           working_directory: /pika/build
           when: always
 
+  arm64_build:
+    <<: *setup_arm64_machine
+    steps:
+      - checkout
+      - run:
+          name: Setup dependencies Arm64
+          command: |
+            # Check architecture = arm
+            uname -m
+            # Install CMake
+            wget -q https://cmake.org/files/v3.22/cmake-3.22.6-linux-aarch64.tar.gz \
+              -O cmake.tar.gz && echo \
+              "79be85d3e76565faacd60695cee11d030f7e7dd8691476144fa25eb93dbd0397  cmake.tar.gz" \
+              | sha256sum --check --status && \
+              sudo tar --strip-components=1 -C /usr/local -xf cmake.tar.gz && \
+              rm cmake.tar.gz
+            cmake --version
+            # Install pika dependencies
+            sudo apt install clang ninja-build libboost-regex-dev \
+              libboost-context-dev libhwloc-dev mpi-default-dev -y
+            mkdir build
+          working_directory: /home/circleci/project
+      - run:
+          name: Clone pika Arm64
+          command: |
+            git clone https://github.com/pika-org/pika.git
+          working_directory: /home/circleci/project
+      - run:
+          name: Configure pika Arm64
+          command: |
+            cmake \
+                /home/circleci/project/pika \
+                -G "Ninja" \
+                -DCMAKE_BUILD_TYPE=Debug \
+                -DCMAKE_CXX_COMPILER=clang++ \
+                -DCMAKE_CXX_FLAGS="-fcolor-diagnostics" \
+                -DPIKA_WITH_GIT_COMMIT=${CIRCLE_SHA1} \
+                -DPIKA_WITH_GIT_BRANCH="${CIRCLE_BRANCH}" \
+                -DPIKA_WITH_GIT_TAG="${CIRCLE_TAG}" \
+                -DPIKA_WITH_CXX_STANDARD=17 \
+                -DPIKA_WITH_PRECOMPILED_HEADERS=On \
+                -DPIKA_WITH_TOOLS=On \
+                -DPIKA_WITH_MALLOC=system \
+                -DPIKA_WITH_MPI=ON \
+                -DPIKA_WITH_EXAMPLES=On \
+                -DPIKA_WITH_GENERIC_CONTEXT_COROUTINES=On \
+                -DPIKA_WITH_TESTS=On \
+                -DPIKA_WITH_TESTS_HEADERS=On \
+                -DPIKA_WITH_PARALLEL_TESTS_BIND_NONE=On \
+                -DPIKA_WITH_COMPILER_WARNINGS=On \
+                -DPIKA_WITH_COMPILER_WARNINGS_AS_ERRORS=On \
+                -DPIKA_WITH_DEPRECATION_WARNINGS=On \
+                -DPIKA_WITH_THREAD_LOCAL_STORAGE=On \
+                -DPIKA_WITH_STACKTRACES_STATIC_SYMBOLS=On \
+                -DPIKA_WITH_STACKTRACES_DEMANGLE_SYMBOLS=Off \
+                -DPIKA_WITH_TESTS_DEBUG_LOG=On \
+                -DPIKA_WITH_TESTS_DEBUG_LOG_DESTINATION=/build/debug-log-arm.txt \
+                -DPIKA_WITH_SPINLOCK_DEADLOCK_DETECTION=On
+          working_directory: /home/circleci/project/build
+      - run:
+          name: Build pika Arm64
+          command: |
+            ninja -j4 -k 0 pika
```

Let's try if this can build without running out of memory:
```suggestion
            ninja -k 0 pika
```